src/controllers/requestController.js

```diff
@@ -1,67 +1,119 @@
+    } catch (error) {
+      return res.status(500).json({ error: error.message });
     }
-    throw new Error('Request not found');
-  } catch (error) {
-    return res.status(500).json({error: "Internal server error!"});
   }
-};
-
-//deleting a request
-const deleteRequest = async (req, res) => {
-  try {
-    const { idRequest } = req.params;
-    const deleted = await model.request.destroy({
-      where: { id: idRequest}
-    });
-    if (deleted) {
-      return res.status(200).json({ message: "Request deleted"});
+
+  // creating a request
+  static async createRequest(req, res) {
+    try {
+      const newIdRoom = req.body.idRoom,
+        allRequests = await findRequestByRoomId(newIdRoom);
+
+      if (allRequests) return res.status(403).json({ message: res.__('Room id is already occupied!') });
+      const {
+          idRoom, dateStart, dateEnd
+        } = req.body,
+        userId = req.userData.id;
+
+      const existingRoom = await findRooms(idRoom);
+
+      if (!existingRoom) return res.status(400).json({ message: res.__('Room id does not exist') });
+
+      const newRequest = {
+        idUser: userId,
+        idRoom,
+        dateStart,
+        dateEnd
+      };
+      const savedRequest = await request.create(newRequest);
+
+      return res.status(201).json({ message: res.__('Request created successfully!'), savedRequest });
+    } catch (error) {
+      return res.status(500).json({ error: res.__('Internal server error!') });
     }
-    throw new Error("Request id provided is not found");
-  } catch (error) {
-    return res.status(500).json({error: "Internal server error!"});
   }
-};
 
-//getting all requests
+  // updating a request
+  static async updateRequest(req, res) {
+    try {
+      const userId = req.userData.id,
+        existingRequests = await findRequestByUserId(userId),
+        userRequestIds = existingRequests.map((item) => item.id);
 
-const getAllRequests = async (req, res) => {
-    const accommodation_requests = await model.request.findAll({});
-    return res.status(200).json({ message: 'Requests found successfully!', accommodation_requests });
-}
+      if (!userRequestIds) return res.status(404).json({ message: 'No request found!' });
+      const requestId = req.params.id,
+        overallRequest = await findRequestById(requestId);
 
-module.exports = {createRequest,updateRequest,deleteRequest,getAllRequests};
+      if (!overallRequest) return res.status(404).json({ message: res.__('Request Id does not exist!') });
+      const matchingRequestId = userRequestIds.find((elem) => elem === overallRequest.id);
+
+      if (!matchingRequestId) return res.status(404).json({ message: res.__('Request does not exist!') });
+      const { requestStatus } = overallRequest;
+
+      if (requestStatus === 'APPROVED') return res.status(403).json({ message: res.__('Request is already APPROVED!') });
+      await request.update(req.body, { where: { id: requestId } });
+
```

Comment:
Here is an issue, you are only checking if the request has been approved it cannot be updated but what if the request was rejected??.... I saw that when it is rejected can also be updated which is not the intended behavior.

here there is a way to fix: 1. You should also check if the `requestStatus` is `REJECTED` to throw error.  Or
                                           2.  Check only if the `requestStatus` is `PENDING` then allow update else throw error...
  The second option is better because it avoids code duplication.

Reply:
ok.